test/adapter/json_api/parse_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          class ParseTest < Minitest::Test
+            def setup
```

Comment:
I kinda disagree. The job of this piece of code is to do a transformation between two specified formats (JSON API on the one side, ActiveRecord#initialize on the other), which is currently being tested. I'm not against an integration test as described in your scenario, though. It just does not seem necessary to me to ensure the validity of this code.